runtime/parachains/src/scheduler.rs

```diff
@@ -203,13 +203,6 @@ decl_module! {
 impl<T: Config> Module<T> {
 	/// Called by the initializer to initialize the scheduler module.
 	pub(crate) fn initializer_initialize(_now: T::BlockNumber) -> Weight {
-		Self::schedule(Vec::new());
-
-		0
-	}
-
-	/// Called by the initializer to finalize the scheduler module.
-	pub(crate) fn initializer_finalize() {
```

Comment:
Removing this logic from `initializer_finalize` is a pretty large change.

Previously, the `Scheduled` paras were never persisted to the merkle trie.

Reply:
Is it that large though?

I mean `Scheduled` is not that giant, at most 5k. That's way below 16k where the size of a kv value starts to influence the load time. It also doesn't grow past that. One downside here I see is that we increase the storage transaction delta and thus increase the burden on archive nodes. However, I think so far we haven't cared about that too much and the rest of the storage changes we do on the course of a block dwarf this one.

The alternative here is to call `schedule(Vec::new())` at the beginning of `validator_groups` and `availability_cores` runtime APIs. That would allow us to save on the archive node burden but that would cost us some processing time.  Hopefully, that would hit the state cache, but even with that there are still some storage requests meaning a round trips to the host. And assuming runtime API requests caching, that happens twice.

Reply:
Counting it up per the docs, I make each `CoreAssignment` to be about 150 bits. IIRC our initial estimates are 100 validators, and 10 validators per core, scaling to 1000 validators and 3 per core. Taking the higher of those estimates, that means an additional 50kb in storage. I don't have a good intuition for how expensive storage costs are, but 50k feels pretty small to me.

Reply:
Math correction: 50 kilo-_bits_. That's about 6.2 kilobytes.

Reply:
FWIW, 5k value comes from my back of the envelope calculations:

The encoded size of `CoreAssignment` was `4 + 4 + 1 + 32 + 4 + 4`, i.e `49`. I also used 100 slots, so that is how I get ~5k. 

AFAIU, we only care here about the core count.

Reply:
yeah, should be fine to commit to storage.